lib/compat/wordpress-6.4/blocks.php

```diff
@@ -0,0 +1,19 @@
+<?php
+/**
+ * Temporary compatibility shims for block APIs present in Gutenberg.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Adds delete_posts capabilities to the wp_block post type.
+ *
+ * @param array $args Register post type args.
+ *
+ * @return array Register post type args.
+ */
+function gutenberg_add_custom_capabilities_to_wp_block( $args ) {
+	$args['capabilities']['delete_posts'] = 'delete_posts';
```

Comment:
Should we add a guard here to protect against some rogue filter removing `capabilities` or making it not an array or something?

Reply:
Good point, thanks. I can't see anywhere where we doing any real sanitization after the [apply_filters call](https://github.com/WordPress/wordpress-develop/blob/9e9887d8b8fa49685c484ee606748e1f0aca1bf9/src/wp-includes/class-wp-post-type.php#L463). It certainly can't hurt. Will update.

Reply:
Just been testing some things and noting them here:

```php
$args['capabilities']['delete_posts'] = 'delete_posts';
print_r($args);
```

outputs

```php
Array
(
    [capabilities] => Array
        (
            [delete_posts] => delete_posts
        )

)
```

Where `$args` is `null`, an empty array or not set at all.

On where `$args` or `$args['capabilities']` is a string/number does it bork so will go ahead and add some defensive turrets 🏰

Reply:
The new fortifications (https://github.com/WordPress/gutenberg/pull/53405/commits/88a09b5d23051e0cbb150ff78f4024bc6ef27c49) look good following what you found above. Thanks.